kotlinx-coroutines-core/build.gradle

```diff
@@ -177,12 +177,26 @@ jvmTest {
     minHeapSize = '1g'
     maxHeapSize = '1g'
     enableAssertions = true
-    systemProperty 'java.security.manager', 'kotlinx.coroutines.TestSecurityManager'
+    if (!Idea.active) {
+        // We should not set this security manager when `jvmTest`
+        // is invoked by IntelliJ IDEA since we need to pass
+        // system properties for Lincheck and stress tests.
+        systemProperty 'java.security.manager', 'kotlinx.coroutines.TestSecurityManager'
+    }
     // 'stress' is required to be able to run all subpackage tests like ":jvmTests --tests "*channels*" -Pstress=true"
     if (!Idea.active && rootProject.properties['stress'] == null) {
+        exclude '**/*LincheckTest.*'
         exclude '**/*StressTest.*'
     }
-    systemProperty 'kotlinx.coroutines.scheduler.keep.alive.sec', '100000' // any unpark problem hangs test
+    if (Idea.active) {
```

Comment:
There is a separate `jvmLincheckTest` task that one can also use from IDEA. Such extensive tweaking of options for the regular `jvmTest` is not a good idea, as makes figuring out what's wrong with regular tests harder. I suggest to simply exclude lincheck tests from `jvmTest` configuration.

Reply:
It is too inconvenient if you cannot simply press the "run" button on your lincheck test. Moreover, using this special `jvmLincheckTest` from IDEA does not seem obvious for me.

Reply:
The only real problem is that we have to switch off our custom security manager for testing. Another option would be allowing it to set the required for Lincheck system properties.